akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -458,6 +498,19 @@ class ShardRegion(
     }
   }
 
+  def receiveRegionStateQuery(ref: ActorRef): Unit = {
+    import akka.pattern.{ ask, pipe }
+    import scala.concurrent.duration._
+    implicit val timeout: Timeout = 1.seconds
+    Future.sequence(shards.map {
+      case (shardId, ref) ⇒
+        (ref ? Shard.GetCurrentShardState).mapTo[Shard.CurrentShardState]
+    }).map { shardStates ⇒
+      CurrentShardRegionState(shardStates.map(state ⇒
+        ShardRegion.ShardState(state.shardId, state.entityIds)).toSet)
+    }.pipeTo(ref)
```

Comment:
Thinking a bit more about it, should that be any kind of failure getting the states? 

I'm thinking log the failure at warning and send that back. a `GetShardRegionStateFailure` since the sender can't do anything about it anyways. Bad idea?